raft/log.go

```diff
@@ -713,24 +737,30 @@ loop:
 
 // leaderLoop periodically sends heartbeats to all followers to maintain dominance.
 func (l *Log) leaderLoop(done chan struct{}) {
-	ticker := l.Clock.Ticker(l.HeartbeatInterval)
-	defer ticker.Stop()
+	l.tracef("leader loop: start")
+	confirm := make(chan struct{}, 0)
 	for {
 		// Send hearbeat to followers.
 		if err := l.sendHeartbeat(done); err != nil {
+			close(confirm)
 			return
 		}
 
+		// Signal clock that the heartbeat has occurred.
+		close(confirm)
+
 		select {
 		case <-done: // wait for state change.
 			return
-		case <-ticker.C: // wait for next heartbeat
+		case confirm = <-l.Clock.AfterHeartbeatInterval(): // wait for next heartbeat
 		}
 	}
 }
 
 // sendHeartbeat sends heartbeats to all the nodes.
 func (l *Log) sendHeartbeat(done chan struct{}) error {
+	l.tracef("sending heartbeat")
```

Comment:
These trace messages are handy, but the text could be deceiving to the next guy. Since it's a log message, it looks like it's stating that something is happening (that is how I read it initially), but really it's just stating that it's in the `sendHeartbeat` function. Perhaps the text could be tweaked to just `sendHeartbeat`, which tends to indicate it's a symbol. Ditto for the rest of the function.

Not a huge deal, but definitely something that jumped out at me.